returnn/tf/util/data.py

```diff
@@ -871,27 +888,46 @@ def translate_axis(axis):
       assert new_axis <= axis < old_axis
       return axis + 1
 
-    data = self.copy()
-    if data.placeholder is not None:
-      if keep_other_axes_intact:
-        data.placeholder = swapaxes(data.placeholder, old_axis, new_axis)
-      else:
-        data.placeholder = move_axis(data.placeholder, old_axis, new_axis)
-    data.batch_dim_axis = translate_axis(self.batch_dim_axis)
-    new_feature_dim_axis = translate_axis(self.feature_dim_axis)
-    if new_feature_dim_axis != data.feature_dim_axis:
-      # Only assign in this case. Otherwise, e.g. if it is NotSpecified, leave it like that.
-      data.feature_dim_axis = new_feature_dim_axis
-    data.time_dim_axis = translate_axis(self.time_dim_axis)
-    if data.size_placeholder:
-      data.size_placeholder = {
-        data.get_batch_axis_excluding_batch(translate_axis(self.get_batch_axis(i))): size
-        for (i, size) in data.size_placeholder.items()}
-      assert None not in data.size_placeholder
-    new_shape = [None] * data.ndim
-    for i, dim in enumerate(self.shape):
-      new_shape[data.get_batch_axis_excluding_batch(translate_axis(self.get_batch_axis(i)))] = dim
-    data.shape = tuple(new_shape)
+    data = self._copy_template_translate_axes(translate_axis)
+    if self.placeholder is not None:
+      from .basic import move_axis
+      data.placeholder = move_axis(self.placeholder, old_axis, new_axis)
+    data.sanity_check()
+    return data
+
+  def copy_swap_axes(self, axis1, axis2):
+    """
+    Like :func:`Data.copy_move_axis`, but keeps all other axes unchanged.
+    :param int axis1: counted with batch-dim
+    :param int axis2: counted with batch-dim
+    :return: copy of myself with moved axis (see :func:`swapaxes`)
+    :rtype: Data
+    """
+    if axis1 < 0:
+      axis1 += self.batch_ndim
+    assert 0 <= axis1 < self.batch_ndim
+    if axis2 < 0:
+      axis2 += self.batch_ndim
+    assert 0 <= axis2 < self.batch_ndim
+    if axis1 == axis2:
+      return self.copy()
+
+    def translate_axis(axis):
+      """
+      :param int|None axis:
+      :return: axis after move_axis
+      :rtype: int|None
+      """
+      if axis == axis1:
+        return axis2
+      if axis == axis2:
+        return axis1
+      return axis
+
+    data = self._copy_template_translate_axes(translate_axis)
+    if self.placeholder is not None:
+      from .basic import swapaxes
+      data.placeholder = swapaxes(self.placeholder, axis1, axis2)
```

Comment:
Same here. I think `tf.transpose` might be easier. But actually, if you directly implement `copy_transpose`, you don't need any additional logic here.